WordPress/gutenberg
# Blocks: Refactor blocks to use supports align

## Description
Follow up for #4069.

This PR seeks the way to refactor all existing core block to use newly introduced `align` option for inside `supports` field. It also adds missing changes to the documentation.

I skipped ~3~ 4 blocks which use non-standard properties:
- image
- paragraph
- text-columns
- latests-posts

We may leave it as it is or we can tackle in another PR.
 
## How to test

For testing purpose it makes sense to enable wide alignment as described [here](https://github.com/WordPress/gutenberg/blob/master/docs/themes.md#wide-alignment).

Verify that there are no regressions in the alignment behavior of the following blocks:
- button
- categories
- columns
- cover image
- embed
- gallery
- ~latest posts~ - removed from this PR, see: https://github.com/WordPress/gutenberg/pull/5099#discussion_r169828258
- pullquote
- table
- video

Ensure unit tests pass:

```
npm test
```
## Types of changes
Refactoring.

## Checklist:
